array.h

```diff
@@ -9,6 +9,9 @@ typedef struct _bdd_array {
 
 _bdd_array* _bdd_array_create() {
     _bdd_array *arr = malloc(sizeof(_bdd_array));
+    if (!arr) {
+        return NULL;
```

Comment:
I think if we can not `malloc` we should just exit with a non-zero status code. Maybe also with a message, although I'm not sure you can output to console it malloc is failing...

Reply:
I think perror + abort would be okay here -- worst case, you don't see a message and the process still aborts.